### PR TITLE
FTL

### PR DESCRIPTION
**Pull request by DL6ER**
**By submitting this pull request, I confirm the following (please check boxes, eg [X]) _Failure to fill the template will close your PR_:**

***Please submit all pull requests against the `development` branch. Failure to do so will delay or deny your request***

- [X] I have read and understood the [contributors guide](https://github.com/pi-hole/pi-hole/blob/master/CONTRIBUTING.md).
- [X] I have checked that [another pull request](https://github.com/pi-hole/pi-hole/pulls) for this purpose does not exist.
- [X] I have considered, and confirmed that this submission will be valuable to others.
- [X] I accept that this submission may not be used, and the pull request closed at the will of the maintainer.
- [X] I give this submission freely, and claim no ownership to its content.

**How familiar are you with the codebase?:** 10

---

FTL integration into installer

_This template was created based on the work of [`udemy-dl`](https://github.com/nishad/udemy-dl/blob/master/LICENSE)._

### CODE REVIEW

File: automated install/basic-install.sh [hunk 1]
```diff
@@ -1150,6 +1156,66 @@ if [[ "${reconfigure}" == true ]]; then
...
+    # 64bit
+    echo ":::  Detected x86_64 architecture"
+    binary="pihole-FTL-linux-x86_64"
+  else
+    # Something else - we try to use 32bit executable and warn the user
+    if [[ ! $machine == i686 ]]; then
+      echo ":::  Not able to detect architecture (unknown: ${machine}), trying 32bit executable"
+      echo ":::  Contact Pi-hole support if you experience problems (like FTL not running)"
+    fi
+    binary="pihole-FTL-linux-x86_32"
```
Comment (Mcat12) [inline automated install/basic-install.sh]: How about check for 32 bit first (and say that it's detected), but if it's still not known default to 32 bit and show the message. Then the 32 bit users will also have the `:::  Detected x86_64 architecture` line but unknown ones will still function the same.

File: automated install/basic-install.sh [hunk 2]
```diff
@@ -1150,6 +1156,66 @@ if [[ "${reconfigure}" == true ]]; then
...
+    return 0
+  else
+    echo ":::   failed (download of binary from Github failed)"
+    return 1
+  fi
+}
+
+FTLinstall() {
+  # Download suitable FTL binary
+  echo ":::"
```
Comment (Mcat12) [inline automated install/basic-install.sh]: Download -> Install

**Comment by dschaper** [comment]
Adding to test rig for automated testing of binary selection.

**Comment by dschaper** [comment]
```
Last login: Mon Feb 20 05:33:00 2017 from 192.168.1.246
root@odroid64:~# uname -a
Linux odroid64 3.14.79-105 #1 SMP PREEMPT Tue Jan 31 21:16:32 BRST 2017 aarch64 aarch64 aarch64 GNU/Linux
root@odroid64:~# uname -m
aarch64
root@odroid64:~# cat /etc/lsb-release 
DISTRIB_ID=Ubuntu
root@odroid64:~# 
```

**Comment by dschaper** [comment]
Platform detection passes. Need to refactor function to split platform detection from engine download.

**Comment by dschaper** [comment]
`.port` and `.pid` are still on `/run` which means they won't survive a reboot. But everything is behaving according to current design.

**Comment by dschaper** [comment]
Approved

**Comment by PromoFaux** [comment]
Approved

**Pull request merged**